sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies.py

```diff
@@ -241,6 +243,27 @@ def on_response(self, request, response, **kwargs):
                 _LOGGER.debug("Failed to log response: %s", repr(err))
 
 
+class StorageUserAgentPolicy(SansIOHTTPPolicy):
+
+    _USERAGENT = "User-Agent"
+
+    def __init__(self, **kwargs):
+        self._application = kwargs.pop('user_agent', None)
+        self._user_agent = "azsdk-python-storage.blob/{} Python/{} ({})".format(
+            VERSION,
+            platform.python_version(),
+            platform.platform())
+        super(StorageUserAgentPolicy, self).__init__()
+
+    def on_request(self, request, **kwargs):
+        app_string = request.context.options.pop('user_agent', None) or self._application
+        if app_string:
```

Comment:
We should prepend the UA string to whatever header value that was already there as opposed to completely replacing it...